nitro-protocol/contracts/MultiAssetHolder.sol

```diff
@@ -49,12 +78,7 @@ contract MultiAssetHolder is IMultiAssetHolder, StatusManager {
         uint256 held = holdings[asset][channelId];
         require(held == expectedHeld, 'held != expectedHeld');
 
-        // require successful deposit before updating holdings (protect against reentrancy)
-        if (asset == address(0)) {
-            require(msg.value == amount, 'Incorrect msg.value for deposit');
-        } else {
-            IERC20(asset).safeTransferFrom(msg.sender, address(this), amount);
-        }
+        IERC20(asset).safeTransferFrom(msg.sender, address(this), amount);
```

Comment:
What happens if a zero address is passed is now? Does it revert?

Reply:
It does revert: `reverted with reason string 'Address: call to non-contract'`